src/ctapipe/monitoring/aggregator.py

```diff
@@ -1,163 +1,402 @@
+    ).tag(config=True)
+
+    chunk_shift = Int(
+        default_value=None,
+        allow_none=True,
+        help=(
+            "Number of events to shift between consecutive chunks. "
+            "If None, chunks do not overlap."
+        ),
     ).tag(config=True)
 
+    def _generate_chunks(self, table) -> Generator[Table, None, None]:
+        """Generate event-count based chunks."""
+        # Handle case where chunk_size is None (entire table)
+        if self.chunk_size is None:
+            yield table
+            return
+
+        # Validate chunk_size vs table length
+        if len(table) < self.chunk_size:
+            raise ValueError(
+                f"Table length ({len(table)}) is less than chunk_size ({self.chunk_size})"
+            )
+
+        # Calculate step size
+        step = self.chunk_shift if self.chunk_shift is not None else self.chunk_size
+
+        # Generate overlapping/non-overlapping chunks
+        for i in range(0, len(table) - self.chunk_size + 1, step):
+            yield table[i : i + self.chunk_size]
+
+        # Handle last chunk for non-overlapping case
+        if self.chunk_shift is None and len(table) % self.chunk_size != 0:
+            # Ensure last chunk has full size by potentially overlapping
+            yield table[-self.chunk_size :]
+
+
+class TimeChunking(BaseChunking):
+    """Divides tables into chunks based on time intervals."""
+
+    chunk_duration = AstroQuantity(
+        physical_type=u.s,
+        default_value=0 * u.s,
+        help="Duration of each time chunk.",
+    ).tag(config=True)
+
+    chunk_shift = AstroQuantity(
+        physical_type=u.s,
+        default_value=0 * u.s,
+        allow_none=True,
+        help=("Time shift between consecutive chunks. If 0, chunks do not overlap."),
+    ).tag(config=True)
+
+    time_tolerance = AstroQuantity(
+        physical_type=u.s,
+        default_value=0.1 * u.s,
+        help=(
+            "Time tolerance for floating point comparisons when determining "
+            "chunk boundaries. Used bidirectionally: prevents generating chunks "
+            "when within tolerance of the end time, and prevents generating "
+            "additional overlapping chunks when remaining time is within tolerance."
+        ),
+    ).tag(config=True)
+
+    def _generate_chunks(self, table) -> Generator[Table, None, None]:
+        """Generate time-based chunks."""
+        # Handle case where chunk_duration is None (entire table)
+        if self.chunk_duration is None:
+            yield table
+            return
+
+        times = table["time"]
+        start_time = times[0]
+        end_time = times[-1]
+        total_duration = end_time - start_time
+
+        # Validate inputs
+        if total_duration < self.chunk_duration:
+            raise ValueError(
+                f"Total duration ({total_duration}) is less than chunk_duration ({self.chunk_duration})"
+            )
+        if self.chunk_duration == 0 * u.s:
```

Comment:
```suggestion
        if self.chunk_duration <= 0 * u.s:
```

Reply:
Is it needed? Does astropy allow negative time? It is AstroQuantity with the physical type of time.